internal/checker/checker.go

```diff
@@ -14746,8 +14746,10 @@ func (c *Checker) resolveESModuleSymbol(moduleSymbol *ast.Symbol, referencingLoc
 	symbol := c.resolveExternalModuleSymbol(moduleSymbol, dontResolveAlias)
 	if !dontResolveAlias && symbol != nil {
 		if !suppressInteropError && symbol.Flags&(ast.SymbolFlagsModule|ast.SymbolFlagsVariable) == 0 && ast.GetDeclarationOfKind(symbol, ast.KindSourceFile) == nil {
-			compilerOptionName := core.IfElse(c.moduleKind >= core.ModuleKindES2015, "allowSyntheticDefaultImports", "esModuleInterop")
-			c.error(referencingLocation, diagnostics.This_module_can_only_be_referenced_with_ECMAScript_imports_Slashexports_by_turning_on_the_0_flag_and_referencing_its_default_export, compilerOptionName)
+			if ast.GetDeclarationOfKind(symbol, ast.KindJSExportAssignment) == nil {
```

Comment:
The original code didn't do something like this; @sandersn is this because the JS the external module symbol isn't the source file anymore?

Reply:
I don't understand how this fix is related to the previous code or to the repro in the bug. If you want to know if the file is a commonjs module, there's still CommonJSModuleIndicator in Corsa. But there's nothing about that in the Strada code either.

Reply:
Currently the repro returns nil for `ast.GetDeclarationOfKind(symbol, ast.KindSourceFile)` but (I think) is valid code.

The bug is that, in the repro,

```js
import { foo } from "./shared.vars";
```

fails while

```js
import f from "./shared.vars";
const { foo } = f;
```

is fine

Reply:
Yep, @jakebailey is right on both counts. In Strada `module.exports` merges with the source file, but `export=` does not. It is flagged as Property. In Corsa, that's also how `module.exports` behaves.

And it does make sense to add this check to the previous `if` since it is one of the exemptions to the error.